tests/expressions/tests.py

```diff
@@ -2849,3 +2850,25 @@ def test_nulls_false(self):
             F("field").asc(nulls_first=False)
         with self.assertRaisesMessage(ValueError, msg):
             F("field").desc(nulls_last=False)
+
+
+class SubqueryTests(TestCase):
+    def test_outer_ref_order_by(self):
+        NamedCategory.objects.create(id=1, name="first")
+        NamedCategory.objects.create(id=4, name="fourth")
+        NamedCategory.objects.create(id=2, name="second")
+        NamedCategory.objects.create(id=3, name="third")
+        outer_query = NamedCategory.objects.all()
+
+        subquery = (
+            NamedCategory.objects.filter(id=OuterRef("id"))
+            .order_by("name")
```

Comment:
> Can you please confirm if the OuterRef should be used with the id field or the pk field for the ordering?

It should be "pk" and should also be inside the `order_by` call like this:
```suggestion
            .order_by(OuterRef("pk"))
```

In further review rounds we might polish this test a little further, but this should be enough to reproduce the reported issue 👍